frame/democracy/src/benchmarking.rs

```diff
@@ -0,0 +1,383 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Democracy pallet benchmarking.
+
+use super::*;
+
+use frame_benchmarking::{benchmarks, account};
+use frame_support::traits::{Currency, Get};
+use frame_system::{RawOrigin, Module as System, self};
+use pallet_collective::{RawOrigin as CollectiveOrigin, Instance1, Instance2};
```

Comment:
and benchmark should be related to the node benchmarked not to our node in substrate.
Thus any Origin can be used here, not necessary collective.

I didn't dive into implementation but for me we should bound new trait on our origins associeted types like this:
```rust
	type CancellationOrigin: EnsureOrigin<Self::Origin> + BenchmarkOriginProvider<Self::Origin>;
```
(or directly into EnsureOrigin traits.)
```rust
trait BenchmarkOriginProvider {
  fn random_successful_origin() -> OriginKind;
  // And maybe this one but i'm not sure it is needed.
  fn successful_origin_from_success(success: EnsureOrigin::Success) -> OriginKind;
}
```

Reply:
Yeah, I was thinking something like that, using collective is awful and `Root` doesn't work always. :+1:

Reply:
or maybe something like what is done in https://github.com/paritytech/substrate/pull/5287 for the trait Contains